tests/src/test/java/io/apicurio/tests/BaseIT.java

```diff
@@ -113,4 +122,52 @@ protected void deleteMultipleArtifacts(Map<String, String> idMap) {
             LOGGER.info("Deleted artifact {} with ID: {}", entry.getKey(), entry.getValue());
         }
     }
+
+    public void createArtifactViaApicurioClient(Schema schema, String artifactName) {
+        CompletionStage<ArtifactMetaData> csa = apicurioService.createArtifact(
+                ArtifactType.AVRO,
+                artifactName,
+                new ByteArrayInputStream(schema.toString().getBytes())
+        );
+        ArtifactMetaData artifactMetadata = ConcurrentUtil.result(csa);
+        EditableMetaData editableMetaData = new EditableMetaData();
+        editableMetaData.setName(artifactName);
+        apicurioService.updateArtifactMetaData(artifactName, editableMetaData);
+        // wait for global id store to populate (in case of Kafka / Streams)
+        ArtifactMetaData metadata = apicurioService.getArtifactMetaDataByGlobalId(artifactMetadata.getGlobalId());
```

Comment:
This can return 404 with some storage impls, e.g. Streams - since globalId is a 2-step async process, so you can already get newly created metadata, but the globalId -> artifact mapping is not yet in place.
So this needs to be done in a retry.